onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -3931,25 +3931,31 @@ def aten_negative(self: TensorType) -> TensorType:
 
 
 @torch_op("aten::new_empty")
-def aten_new_empty(self: TTensor, size: INT64) -> TTensor:
+def aten_new_empty(self: TTensor, size: INT64, dtype: int = -1) -> TTensor:
     # new_empty(Tensor self, SymInt[] size, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
 
     # using zero to simulate empty array
     zero = op.Constant(value_float=0.0)
     result = op.Expand(zero, size)
-    result = op.CastLike(result, self)
+    if dtype == -1:
+        result = op.CastLike(result, self)
+    else:
+        result = op.Cast(result, to=dtype)
```

Comment:
dtype needs to be the same for both branches. this op may need to be trace only

Reply:
not got your point

Reply:
So ONNX requires that both branches of an If node’s type to be the same. Since we are casting here the graph violates this constraint. I think Rama is looking at potential solutions, but for now we will need to mark the function traceonly, or else the graph would be invalid.

Reply:
More info in https://github.com/onnx/onnx/issues/4872